pets/mongodb/init/on-start.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function join {
+    local IFS="$1"; shift; echo "$*";
+}
+
+HOSTNAME=$(hostname)
+# Parse out cluster name, formatted as: petset_name-index
+IFS='-' read -ra ADDR <<< "$(hostname)"
+CLUSTER_NAME="${ADDR[0]}"
+
+while read -ra LINE; do
+    if [[ "${LINE}" == *"${HOSTNAME}"* ]]; then
+        MY_NAME=$LINE
+        continue
+    fi
+    PEERS=("${PEERS[@]}" $LINE)
+done
+
+apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 7F0CEB10 \
+ && echo 'deb http://downloads-distro.mongodb.org/repo/ubuntu-upstart dist 10gen' > /etc/apt/sources.list.d/mongodb.list \
+ && apt-get update \
+ && DEBIAN_FRONTEND=noninteractive apt-get install -y mongodb-org-server mongodb-org-shell \
+ && rm -rf /var/lib/apt/lists/*
```

Comment:
If theres an existing mongo image you can simply use it as the base. I might've done this in places that don't have a docker image, but it's a pain to continuously do this for every pod. I ended up baking this into the image we use in e2es for redis and mysql: https://github.com/kubernetes/kubernetes/tree/master/test/images/pets/redis. 

It's preferable to not use a custom image in the app contaienr, inita containers less so.